homeassistant/components/directv/config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Config flow for DirecTV."""
+import logging
+import socket
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from DirectPy import DIRECTV
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_SERIAL
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DEFAULT_PORT
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+ERROR_CANNOT_CONNECT = "cannot_connect"
+ERROR_UNKNOWN = "unknown"
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+def get_ip(host: str) -> str:
+    """Translate hostname to IP address."""
+    if host is None:
```

Comment:
When can this happen ? And shouldn't we just not call this function if it's None?